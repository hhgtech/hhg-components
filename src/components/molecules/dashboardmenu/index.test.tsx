import React from 'react'
import { render, cleanup, screen } from '@testing-library/react'
import 'jest-styled-components'

import { DashboardMenu } from '.'

afterEach(cleanup)

test('it renders', () => {
  const dashboardMenu = render(
    <DashboardMenu
      initalPage={1}
      pages={[
        {
          name: 'Overview',
          icon: <img src="https://source.unsplash.com/random/32x32" />,
          activeIcon: <img src="https://source.unsplash.com/random/32x32" />,
        },

        {
          name: 'Facilities',
          icon: <img src="https://source.unsplash.com/random/32x32" />,
          activeIcon: <img src="https://source.unsplash.com/random/32x32" />,
        },

        {
          name: 'Doctors',
          icon: <img src="https://source.unsplash.com/random/32x32" />,
          activeIcon: <img src="https://source.unsplash.com/random/32x32" />,
        },
      ]}
    />,
  )

  const self = dashboardMenu.container.firstChild

  expect(self).toMatchSnapshot()

  expect(self).toContainHTML('<img')
  expect(self).toHaveTextContent('Overview')
})
