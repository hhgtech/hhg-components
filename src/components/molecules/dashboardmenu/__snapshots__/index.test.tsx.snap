// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c3 {
  color: #262626;
}

.c3[data-type='bold'],
.c3[data-variant='heading'],
.c3[data-type='caption'] {
  font-weight: 700;
}

.c3[data-type='caption'] {
  text-transform: uppercase;
}

.c0 p {
  color: $gray-800;
}

.c0[data-is-active] {
  background: $primary-base;
}

.c0[data-is-active] p {
  color: $white;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 9px 24px;
  margin-bottom: 12px;
  border-radius: 6px;
  cursor: pointer;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-width: 25px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 12px;
}

@media (min-width:768px) {
  .c3[data-size='md'] {
    font-size: 20px;
    -webkit-letter-spacing: -0.4px;
    -moz-letter-spacing: -0.4px;
    -ms-letter-spacing: -0.4px;
    letter-spacing: -0.4px;
    line-height: 1.2;
  }

  .c3[data-size='base'] {
    font-size: 16px;
    line-height: 1.5;
  }

  .c3[data-size='sm'] {
    font-size: 13px;
    line-height: 1.5;
  }

  .c3[data-size='xs'] {
    font-size: 11px;
    line-height: 1.5;
  }
}

@media (max-width:767px) {
  .c3[data-size='md'] {
    font-size: 16px;
    line-height: 1.2;
  }

  .c3[data-size='base'] {
    font-size: 14px;
    line-height: 1.5;
  }

  .c3[data-size='sm'] {
    font-size: 13px;
    line-height: 1.5;
  }

  .c3[data-size='xs'] {
    font-size: 11px;
    line-height: 1.5;
  }
}

<div
  class="c0"
>
  <ul>
    <li
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          src="https://source.unsplash.com/random/32x32"
        />
      </div>
      <p
        class="c3"
        data-size="base"
        data-type="bold"
      >
        Overview
      </p>
    </li>
    <li
      class="c1"
      data-is-active="true"
    >
      <div
        class="c2"
      >
        <img
          src="https://source.unsplash.com/random/32x32"
        />
      </div>
      <p
        class="c3"
        data-size="base"
        data-type="bold"
      >
        Facilities
      </p>
    </li>
    <li
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          src="https://source.unsplash.com/random/32x32"
        />
      </div>
      <p
        class="c3"
        data-size="base"
        data-type="bold"
      >
        Doctors
      </p>
    </li>
  </ul>
</div>
`;
